documentation/specs/src/interoperability/ethereum-bridge/ethereum_events_attestation.md

```diff
@@ -18,15 +21,26 @@ pub enum EthereumEvent {
 }
 ```
 
-Each event will be stored with a list of the validators that have ever seen it
-as well as the fraction of total voting power that has ever seen it.
-Once an event has been seen by 2/3 of voting power, it is locked into a
+Each event will be stored with a list of the consensus validators that have
+ever seen it as well as the fraction of total voting power that has ever seen it.
+Once an event has been seen by at least 2/3 of voting power, it is locked into a
 `seen` state, and acted upon.
 
-There is no adjustment across epoch boundaries - e.g. if an event is seen by 1/3
-of voting power in epoch n, then seen by a different 1/3 of voting power in
-epoch m>n, the event will be considered `seen` in total. Validators may never
-vote more than once for a given event.
+If the voting power of Namada changes across epoch boundaries, then events in
+storage which are yet to be achieve a quorum decision behind them (i.e. whose
+`seen` state is still `false`) must have their voting power adjusted. It is
+enough to lazily adjust an event's voting power whenever a new vote is made
+for it, to avoid iterating over each Ethereum event in storage.
+
+Validators may never vote more than once for a given event. To ensure that this
+invariant is held, events are timed out if they are not `seen` within the span
```

Comment:
We also check that a specific validator hasn't voted for a specific event more than once, right?

Reply:
yeah. `seen_by`, in storage, tracks who voted on some event. it's a function from a validator's address to the height in storage at which they voted on some eth event